modules/command-engine/core/src/main/scala/surge/internal/persistence/KTablePersistenceSupport.scala

```diff
@@ -62,30 +75,43 @@ trait KTablePersistenceSupport[Agg, Event] {
       serializedState: KafkaProducerActor.MessageToPublish,
       currentFailureCount: Int = 0,
       startTime: Instant,
-      didStateChange: Boolean)(implicit ec: ExecutionContext): Future[Any] = {
+      didStateChange: Boolean,
+      trackingId: UUID = UUID.randomUUID())(implicit ec: ExecutionContext): Future[Any] = {
     log.trace("Publishing messages for {}", aggregateId)
     if (serializedEvents.isEmpty && !didStateChange) {
-      Future.successful(PersistenceSuccess(state, startTime, context))
+      Future.successful(PersistenceSuccess(trackingId, state, startTime, context))
     } else {
-      kafkaProducerActor
-        .publish(aggregateId = aggregateId, state = serializedState, events = serializedEvents)
+      val futureResult = kafkaProducerActor.publish(aggregateId = aggregateId, state = serializedState, events = serializedEvents, requestId = trackingId)
+      futureResult
         .map {
-          case KafkaProducerActor.PublishSuccess => PersistenceSuccess(state, startTime, context)
-          case KafkaProducerActor.PublishFailure(t) =>
-            PersistenceFailure(state, context, t, currentFailureCount + 1, serializedEvents, serializedState, startTime)
+          case KafkaProducerActor.PublishSuccess(publishRequestId) => PersistenceSuccess(publishRequestId, state, startTime, context)
+          case KafkaProducerActor.PublishFailure(publishRequestId, t, _) =>
+            PersistenceFailure(publishRequestId, state, context, t, currentFailureCount + 1, serializedEvents, serializedState, startTime)
         }
         .recover { case t =>
           log.error("Failed to publish messages", t)
-          EventPublishTimedOut(t, startTime)
+
+          t match {
+            case exception: RetryAwareException =>
```

Comment:
I think we should track `currentFailureCount` here as well to put a ceiling on how many attempts before we give up

Reply:
done